src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -140,6 +141,418 @@ impl<'a, P: PageTableFrameMapping> MappedPageTable<'a, P> {
+                );
+                let p4_start = p4_start.max(pages.start);
+                let p4_end = Page::from_page_table_indices_2mib(
+                    p4_index,
+                    PageTableIndex::new(511),
+                    PageTableIndex::new(511),
+                );
+                let p4_end = p4_end.min(pages.end);
+
+                if p4_start == p4_end {
+                    return Ok(());
+                }
+
+                let p3 = next_table(&mut info, &mut p4[p4_index], page_table_walker)
+                    .map_err(|e| (e.into(), p4_start))?;
+
+                let start_p3_index = p4_start.p3_index();
+                let end_p3_index = p4_end.p3_index();
+
+                (start_p3_index.into()..=end_p3_index.into())
+                    .map(PageTableIndex::new)
+                    .try_for_each(|p3_index| {
+                        let p3_start = Page::from_page_table_indices_2mib(
+                            p4_index,
+                            p3_index,
+                            PageTableIndex::new(0),
+                        );
+                        let p3_start = p3_start.max(p4_start);
+                        let p3_end = Page::from_page_table_indices_2mib(
+                            p4_index,
+                            p3_index,
+                            PageTableIndex::new(511),
+                        );
+                        let p3_end = p3_end.min(p4_end);
+
+                        if p3_start == p3_end {
+                            return Ok(());
+                        }
+
+                        let p2 = next_table(&mut info, &mut p3[p3_index], page_table_walker)
+                            .map_err(|e| (e.into(), p3_start))?;
+
+                        let start_p2_index = p3_start.p2_index().into();
+                        let mut end_p2_index = p3_end.p2_index().into();
+
+                        if p3_end != pages.end {
+                            end_p2_index += 1;
+                        }
+
+                        (start_p2_index..end_p2_index)
+                            .map(PageTableIndex::new)
+                            .map(move |p2_index| {
+                                Page::from_page_table_indices_2mib(p4_index, p3_index, p2_index)
+                            })
+                            .try_for_each(|page| {
+                                let entry = &mut p2[page.p2_index()];
+                                modify(entry, page, &mut info).map_err(|e| (e, page))
+                            })
+                    })
+            })
+            .map(|_| MapperFlushRange::new(pages))
+            .map_err(|(e, page)| {
+                (
+                    e,
+                    MapperFlushRange::new(PageRange {
+                        start: pages.start,
+                        end: page,
+                    }),
+                )
+            })
+    }
+
+    #[inline]
+    fn map_range_2mib<F, A>(
```

Comment:
Perhaps call this `map_to_range_2mib` to be consistent with the other functions?

Reply:
Yes